runtime/queries/column_timeseries.go

```diff
@@ -103,24 +104,30 @@ func (q *ColumnTimeseries) Resolve(ctx context.Context, rt *runtime.Runtime, ins
 		}
 
 		measures := normaliseMeasures(q.Measures, q.Pixels != 0)
-		dateTruncSpecifier := convertToDateTruncSpecifier(timeRange.Interval)
+		timeBucketSpecifier := convertToTimeBucketSpecifier(timeRange.Interval)
 		tsAlias := tempName("_ts_")
 		temporaryTableName := tempName("_timeseries_")
+
+		timezone := "UTC"
+		if q.TimeZoneAdjustment != nil && q.TimeZoneAdjustment.GetTimeZone() != "" {
+			timezone = q.TimeZoneAdjustment.GetTimeZone()
+		}
 		querySQL := `CREATE TEMPORARY TABLE ` + temporaryTableName + ` AS (
 			-- generate a time series column that has the intended range
 			WITH template as (
 			SELECT
-				range as ` + tsAlias + `
+				timezone('` + timezone + `',range) as ` + tsAlias + `
 			FROM
 				range(
-				date_trunc('` + dateTruncSpecifier + `', TIMESTAMP '` + timeRange.Start.AsTime().Format(IsoFormat) + `'),
-				date_trunc('` + dateTruncSpecifier + `', TIMESTAMP '` + timeRange.End.AsTime().Format(IsoFormat) + `'),
-				INTERVAL '1 ` + dateTruncSpecifier + `')
+				time_bucket(INTERVAL '` + timeBucketSpecifier + `', TIMESTAMP '` + timeRange.Start.AsTime().Format(IsoFormat) + `'),
+				time_bucket(INTERVAL '` + timeBucketSpecifier + `', TIMESTAMP '` + timeRange.End.AsTime().Format(IsoFormat) + `'),
+				INTERVAL '` + timeBucketSpecifier + `')
```

Comment:
I see, although it requires double conversion otherwise it shows the non-existent hour, ie:
```sql
SELECT time_bucket(interval '1 hour', range::timestamptz, 'America/St_Johns')                                 bucket_in_utc,
       time_bucket(interval '1 hour', range::timestamptz, 'America/St_Johns') at time zone 'America/St_Johns' bucket_in_stjohns,
       range                                                                                                  generated_time
FROM   range(timestamp '2023-03-12 05:00:00', timestamp '2023-03-12 08:00:00', interval '1 hour');
```
```
┌──────────────────────────┬─────────────────────┬─────────────────────┐
│      bucket_in_utc       │  bucket_in_StJohns  │   generated_time    │
│ timestamp with time zone │      timestamp      │      timestamp      │
├──────────────────────────┼─────────────────────┼─────────────────────┤
│ 2023-03-12 04:30:00+00   │ 2023-03-12 01:00:00 │ 2023-03-12 05:00:00 │
│ 2023-03-12 05:30:00+00   │ 2023-03-12 03:00:00 │ 2023-03-12 06:00:00 │
│ 2023-03-12 06:30:00+00   │ 2023-03-12 04:00:00 │ 2023-03-12 07:00:00 │
└──────────────────────────┴─────────────────────┴─────────────────────┘
```

Reply:
```sql
SELECT   Count(*),
         time_bucket(interval '1 hour', range::timestamptz, 'America/St_Johns') at time zone 'America/St_Johns' bucket_in_stjohns
FROM     range(timestamp '2023-11-05 03:00:00', timestamp '2023-11-05 08:00:00', interval '1 hour')
GROUP BY bucket_in_stjohns;
```
```
┌──────────────┬─────────────────────┐
│ count_star() │  bucket_in_StJohns  │
│    int64     │      timestamp      │
├──────────────┼─────────────────────┤
│            1 │ 2023-11-05 00:00:00 │
│            2 │ 2023-11-05 01:00:00 │
│            1 │ 2023-11-05 02:00:00 │
│            1 │ 2023-11-05 03:00:00 │
└──────────────┴─────────────────────┘
```

Reply:
This is for backward DST 👆

Reply:
while in UTC it will look fine:
```sql
SELECT   Count(*),
         time_bucket(interval '1 hour', range::timestamptz, 'America/St_Johns') bucket_in_utc
FROM     range(timestamp '2023-11-05 03:00:00', timestamp '2023-11-05 08:00:00', interval '1 hour')
GROUP BY bucket_in_utc;
```
```
┌──────────────┬──────────────────────────┐
│ count_star() │      bucket_in_utc       │
│    int64     │ timestamp with time zone │
├──────────────┼──────────────────────────┤
│            1 │ 2023-11-05 02:30:00+00   │
│            1 │ 2023-11-05 03:30:00+00   │
│            1 │ 2023-11-05 04:30:00+00   │
│            1 │ 2023-11-05 05:30:00+00   │
│            1 │ 2023-11-05 06:30:00+00   │
└──────────────┴──────────────────────────┘
```